### PR TITLE
HPCC-12885 Introduce backward compatible statistics

### PR DESCRIPTION
**Pull request by ghalliday**
Signed-off-by: Gavin Halliday gavin.halliday@lexisnexis.com

### CODE REVIEW

**Comment by ghalliday** [comment]
@richardkchapman @jakesmith please review.

At the moment this change outputs statistics to the progress graph in the legacy and new formats.  This has the advantage of allowing old graph controls to display skew etc properly, but also ensures the stats are available in their new consistent format.  The disadvantage is more attributes set on the graph (but not stored in the workunit).

**Comment by ghalliday** [comment]
I started adding some code in a separate commit to conditionally include/remove the legacy tags, but I'm not convinced it is worth it.  What do you think?

**Comment by ghalliday** [comment]
(I suspect not - it adds complication to the code which will just be removed again in 6.0).

**Comment by richardkchapman** [comment]
I think I concur

File: system/jlib/jstats.cpp [hunk 1]
```diff
@@ -511,7 +520,7 @@ static const StatisticMeta statsMetaData[StMax] = {
 
 StatisticMeasure queryMeasure(StatisticKind kind)
 {
-    unsigned varient = (kind & ~StKindMask);
+    unsigned varient = queryStatsVarient(kind);
```
Comment (richardkchapman) [inline system/jlib/jstats.cpp]: Not new, but these should be spelled variant rather than varient....

Comment (richardkchapman) [inline system/jlib/jstats.cpp]: Well, half of it is new...

**Comment by richardkchapman** [comment]
@ghalliday minor comments only

**Comment by ghalliday** [comment]
@richardkchapman fixed

**Pull request merged**